modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,315 @@
+
+        }
+        assignPageToExperimentsList(batchExperiments, partialpage);
+
+        // the assignment will happen until the max users cap is reached for all experiments in batch
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            List<Assignment> assignments = postAssignments(application, partialpage, new User("user" + i));
+            assertAssignmentResponseStatus(assignments, "NEW_ASSIGNMENT");
+        }
+
+        // if we do one more assignment the rapid experiments should change to PAUSED/STOPPED state and the remaining
+        // should be in running state
+        List<Assignment> assignments = postAssignments(application, partialpage, new User("user1000"));
+
+        for (Experiment experiment : batchExperiments) {
+            Experiment exp = getExperiment(experiment);
+            if (exp.isRapidExperiment)
+                Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+            else
+                Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_RUNNING);
+        }
+
+    }
+
+    /**
+     * This test case covers a scenario where we do batch assignment for a group of experiments that has all rapid
+     * experiments. It validates the behavior when we do bulk assignment of a user to that batch
+     */
+    @Test
+    public void testAssignmentForBatchWithAllRapidExperiments() {
+
+        page = new Page(BATCH_PAGE, true);
+        List<Experiment> experimentList = new ArrayList<Experiment>();
+        // lets create rapid experiments for the batch
+        for (int i = 1; i <= NUMBER_OF_EXPERIMENTS_PER_BATCH; i++) {
+
+            Experiment exp = ExperimentFactory
+                    .createRapidExperiment(RAPID_EXP_MAX_USERS, "_" + UUID.randomUUID().toString(), -1)
+                    .setApplication(application);
+
+            exp = createBucketsAndRunExperiment(exp, NUMBER_OF_BUCKETS_PER_EXPERIMENT);
+            experimentList.add(exp);
+        }
+
+        assignPageToExperimentsList(experimentList, page);
+
+        // the assignment will happen until the max users cap is reached for all experiments in batch
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            List<Assignment> assignments = postAssignments(application, page, new User("user" + (i + 1000)));
+            assertAssignmentResponseStatus(assignments, "NEW_ASSIGNMENT");
+        }
+
+        // if we do one more assignment the rapid experiments should change to PAUSED/STOPPED state
+        List<Assignment> assignments = postAssignments(application, page, new User("user100"));
+        for (Experiment experiment : experimentList) {
+            Experiment exp = getExperiment(experiment);
+            Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+        }
+
+        // assert that the response of the post assignments are EXPERIMENT_PAUSED
+        assertAssignmentResponseStatus(assignments, "EXPERIMENT_PAUSED");
+
+    }
+
+    /**
```

Comment:
Do you think, it would be useful for all below 4 methods to move to TestBase class so that they can used used across other test classes when needed?